test/components/numericAxisTests.ts

```diff
@@ -427,4 +427,48 @@ describe("NumericAxis", () => {
 
     svg.remove();
   });
+
+  it("constrained tick labels do not overlap tick marks", () => {
+
+    var svg = generateSVG(300, 400);
+
+    var yScale = new Plottable.Scale.Linear().numTicks(100);
+    yScale.domain([175, 185]);
+    var yAxis = new Plottable.Axis.Numeric(yScale, "left")
+                                  .tickLabelPosition("top")
+                                  .tickLength(50);
+
+    var chartTable = new Plottable.Component.Table([
+      [yAxis],
+    ]);
+
+    chartTable.renderTo(svg);
+
+    var tickLabels = (<any> yAxis)._element.selectAll("." + Plottable.Axis.AbstractAxis.TICK_LABEL_CLASS)
+        .filter(function(d: any, i: number) {
+          var visibility = d3.select(this).style("visibility");
+          return (visibility === "visible") || (visibility === "inherit");
+        });
+
+    var tickMarks = (<any> yAxis)._element.selectAll("." + Plottable.Axis.AbstractAxis.TICK_MARK_CLASS)
+        .filter(function(d: any, i: number) {
+          var visibility = d3.select(this).style("visibility");
+          return (visibility === "visible") || (visibility === "inherit");
+        });
+
+
+    tickLabels.each(function() {
+      var tickLabelBox = this.getBoundingClientRect();
+
+      tickMarks.each(function() {
+        var tickMarkBox = this.getBoundingClientRect();
+          assert.isFalse(Plottable._Util.DOM.boxesOverlap(tickLabelBox, tickMarkBox));
```

Comment:
The last argument in any `assert` method is a message. Usually we try to write something detailed there; consider another developer who is working on code that causes this test to fail. They see this:

`Expected true to be false`

with no explanation as to what went wrong. A detailed message provides context when the test fails:

`Expected true to be false: tick label does not overlap a tick mark`